service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -198,18 +198,15 @@ public static void validateBucketName(
     }
   }
 
-  public static void validateControlledResourceRegionAgainstPolicy(
-      TpsApiDispatch tpsApiDispatch, UUID workspaceUuid, String location, CloudPlatform platform) {
+  public static void validateRegionAgainstPolicy(
+      TpsApiDispatch tpsApiDispatch, UUID workspaceUuid, String region, CloudPlatform platform) {
     switch (platform) {
-      case AZURE -> {
-        // TODO: enable policy check in Azure when we support Azure regions in the TPS ontology.
-        // validateAzureRegion(location);
-      }
-      case GCP -> validateGcpRegion(tpsApiDispatch, workspaceUuid, location);
+      case AZURE -> validateRegion(tpsApiDispatch, workspaceUuid, region, platform);
+      case GCP -> validateRegion(
+          tpsApiDispatch, workspaceUuid, GcpUtils.parseRegion(region), platform);
       case ANY -> {
         // Flexible resources are not stored on the cloud. Thus, they have no region policies.
       }
-      default -> throw new InvalidControlledResourceException("Unrecognized platform");
```

Comment:
I removed this because a) it is not required and b) I changed `InvalidControlledResourceException` into a more specific exception: `RegionNotAllowedException`